### PR TITLE
Closes #510: Use organization alternative names in voters

### LINKED ISSUE #510: Use organization alternative names in voters

### PR DESCRIPTION
**Pull request by lukdumi**

### CODE REVIEW

File: iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/FittingAffOrgSectionWordsMatchVoter.java [hunk 1]
```diff
@@ -75,25 +79,40 @@ public FittingAffOrgSectionWordsMatchVoter(List<Character> charsToFilter, int wo
     //------------------------ LOGIC --------------------------
     
     /**
-     * Returns true if minFittingOrgWordsPercentage of {@link AffMatchOrganization#getName()} words
+     * Returns true if minFittingOrgWordsPercentage the words of at least one of the organization names
```
Comment (marekhorst) [inline iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/FittingAffOrgSectionWordsMatchVoter.java]: Is `minFittingOrgWordsPercentage` reference to any variable? I couldn't find any of this name.

This is also referred in two other voters: `FittingAffOrgWordsMatchVoter` and `FittingOrgWordsMatchVoter`.

File: iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/FittingAffOrgWordsMatchVoter.java [hunk 2]
```diff
@@ -71,24 +73,47 @@ public FittingAffOrgWordsMatchVoter(List<Character> charsToFilter, int wordToRem
     //------------------------ LOGIC --------------------------
     
     /**
-     * Returns true if minFittingOrgWordsPercentage of {@link AffMatchOrganization#getName()} words
+     * Returns true if minFittingOrgWordsPercentage of the words in at least one of the organization names
```
Comment (marekhorst) [inline iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/FittingAffOrgWordsMatchVoter.java]: As already mentioned above: misterious name.

File: iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/FittingOrgWordsMatchVoter.java [hunk 3]
```diff
@@ -71,23 +73,45 @@ public FittingOrgWordsMatchVoter(List<Character> charsToFilter, int wordToRemove
     //------------------------ LOGIC --------------------------
     
     /**
-     * Returns true if minFittingOrgWordsPercentage of {@link AffMatchOrganization#getName()} words
+     * Returns true if minFittingOrgWordsPercentage of the words in at least one of the organization names
```
Comment (marekhorst) [inline iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/FittingOrgWordsMatchVoter.java]: As already mentioned above: misterious name.

File: iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/GetOrgAlternativeNamesFunction.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+import eu.dnetlib.iis.wf.affmatching.model.AffMatchOrganization;
+
+/**
+ * Function that returns {@link AffMatchOrganization#getAlternativeNames()} of the passed organization.
+ * 
+* @author Łukasz Dumiszewski
+*/
+
+public class GetOrgAlternativeNamesFunction implements Function<AffMatchOrganization, List<String>>, Serializable {
```
Comment (marekhorst) [inline iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/GetOrgAlternativeNamesFunction.java]: I assume this function was implemented in advance to be used in future, right? I don't see any reference to that function other than unit tests.

File: iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/AffOrgMatchVotersFactoryTest.java [hunk 5]
```diff
@@ -113,8 +122,10 @@ public void createSectionedNameLevenshteinCountryLooseMatchVoter() {
         assertInternalVotersCount(voter, 2);
         
         assertTrue(getInternalVoter(voter, 0) instanceof CountryCodeLooseMatchVoter);
-        assertTrue(getInternalVoter(voter, 1) instanceof SectionedNameLevenshteinMatchVoter);
-        
+        AffOrgMatchVoter voter1 = getInternalVoter(voter, 1);
+        assertTrue(voter1 instanceof SectionedNameLevenshteinMatchVoter);
+        assertTrue(Whitebox.getInternalState(voter1, "getOrgNamesFunction") instanceof GetOrgNameFunction);
```
Comment (marekhorst) [inline iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/AffOrgMatchVotersFactoryTest.java]: I can see three newly added lines (above) which are repeated 5 times in different test methods which are quite similar (only classes verified with `instanceof` are different) . Maybe we could make parameterized `assert` method out of that?

File: iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/SectionedNameStrictMatchVoter.java [hunk 6]
```diff
@@ -17,35 +18,48 @@
...
     @Override
-    protected String getAffiliationName(AffMatchAffiliation affiliation) {
-        return affiliation.getOrganizationName();
+    protected boolean containsOrgSection(List<String> affOrgNameSections, String orgNameSection) {
+        return affOrgNameSections.contains(orgNameSection);
     }
+
     
+    @Override
+    protected List<String> getOrganizationNames(AffMatchOrganization organization) {
```
Comment (marekhorst) [inline iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/SectionedNameStrictMatchVoter.java]: Looking at different `Voters` implementations utilizing `getOrgNamesFunction` field I guess we could introduce another `orgNamesFunction`-aware abstract class in hierarchy including:
- `getOrgNamesFunction` field
- `setGetOrgNamesFunction` method
- `getOrganizationNames` method

this way we would have it implemented in single place.

File: iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/GetOrgAlternativeNamesFunction.java (hunk 4 above)
Comment (lukdumi) [inline iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/GetOrgAlternativeNamesFunction.java]: Yes, I use it in #580.

File: iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/AffOrgMatchVotersFactoryTest.java (hunk 5 above)
Comment (lukdumi) [inline iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/AffOrgMatchVotersFactoryTest.java]: OK

File: iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/FittingAffOrgSectionWordsMatchVoter.java (hunk 1 above)
Comment (lukdumi) [inline iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/FittingAffOrgSectionWordsMatchVoter.java]: :+1:

**Pull request closed without merging**